core/optaplanner-core-impl/src/main/java/org/optaplanner/core/impl/solver/ConsumerSupport.java

```diff
@@ -32,7 +32,7 @@
     private final BiConsumer<? super ProblemId_, ? super Throwable> exceptionHandler;
     private final Semaphore activeConsumption = new Semaphore(1);
     private final BestSolutionHolder<Solution_> bestSolutionHolder;
-    private ExecutorService consumerExecutor = Executors.newSingleThreadExecutor();
+    private final ExecutorService consumerExecutor = Executors.newSingleThreadExecutor();
```

Comment:
Why wasn't this`final` before, and how do these changes make it possible now?

Reply:
The method that calls `consumerExecutor.shutdownNow()` used to set the reference to `null`. Now, this method may be called from multiple threads concurrently: after termination and from the outside via `SolverManager.close()`. 

Thus, either we make the method `synchronized` or let the reference be garbage-collected together with its enclosing `ConsumerSupport` and `SolverJob` instance.